src/ip/QosConfig.cc

```diff
@@ -465,64 +467,61 @@ Ip::Qos::Config::parseConfigLine()
  * which means no StoreEntry references. Just a basic char* buffer.
 */
 void
-Ip::Qos::Config::dumpConfigLine(char *entry, const char *name) const
+Ip::Qos::Config::dumpConfigLine(std::ostream &os, const char *name) const
 {
-    char *p = entry;
     if (isHitTosActive()) {
-
-        p += snprintf(p, 11, "%s", name); // strlen("qos_flows ");
-        p += snprintf(p, 4, "%s", "tos");
+        os << name << " tos";
 
         if (tosLocalHit > 0) {
-            p += snprintf(p, 16, " local-hit=0x%02X", tosLocalHit);
+            os << " local-hit=0x" << asHex(tosLocalHit).minDigits(2);
```

Comment:
`%X` uses [uppercase](https://en.cppreference.com/w/c/io/fprintf). The stream default is lowercase. To preserve formatting, we should add .upperCase() to all these asHex() calls, right?

When fixing this, please put upperCase() in front of minDigits() rather than behind it (see commit 8d85b10 in your asHex() PR).

BTW, this is as use case that can benefit greatly from using a local function to avoid all this code duplication. For example:

```C++
template <class Integer>
static auto asQosHex(const Integer n) { return asHex(n).upperCase().minDigits(2); }
```

Reply:
I'll call it asQosConfigHex, but sure